src/test/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributorTest.java

```diff
@@ -397,11 +449,36 @@ public void collectionExerciseStateTransitionException() {
     } catch (CTPException ex) {
       // Do nothing with it, actually want to catch it in SampleUnitDistributor
     }
+    events
+        .get(0)
+        .setTimestamp(new Timestamp(System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(10)));
+    doReturn(events.get(0))
+        .when(eventRepository)
+        .findOneByCollectionExerciseAndTag(collectionExercise, EventService.Tag.go_live.name());
 
     sampleUnitDistributor.distributeSampleUnits(collectionExercise);
 
     verify(publisher, times(2)).sendSampleUnit(any(SampleUnitParent.class));
     verify(sampleUnitGroupRepo, times(2)).saveAndFlush(any(ExerciseSampleUnitGroup.class));
     verify(collectionExerciseRepo, never()).saveAndFlush(any());
   }
+
+  /** Test if go_live date has past at time of validation. */
+  @Test
+  public void changeCollectionExerciseStateToLiveWhenGoLiveDatePast() throws Exception {
+    // Set collection exercise go live date to be in past
+    events
+        .get(0)
+        .setTimestamp(new Timestamp(System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(10)));
+    doReturn(events.get(0))
+        .when(eventRepository)
+        .findOneByCollectionExerciseAndTag(collectionExercise, EventService.Tag.go_live.name());
+
+    sampleUnitDistributor.distributeSampleUnits(collectionExercise);
+
+    verify(publisher, times(2)).sendSampleUnit(any(SampleUnitParent.class));
```

Comment:
You should probably check that the thing you are sending is what you expect i.e.

verify(publisher, times(2)).sendSampleUnit(new SampleUnitParent(expected data));